Sources/WebAuthn/Ceremonies/Shared/AuthenticatorData.swift

```diff
@@ -43,69 +42,69 @@ extension AuthenticatorData {
             remainingCount -= extensionData!.count
         }
 
-        guard remainingCount == 0 else {
-            throw WebAuthnError.leftOverBytesInAuthenticatorData
-        }
+        guard remainingCount == 0
+        else { throw .leftOverBytesInAuthenticatorData }
 
         self.relyingPartyIDHash = relyingPartyIDHash
         self.flags = flags
         self.counter = counter
         self.attestedData = attestedCredentialData
         self.extData = extensionData
-
     }
 
     /// Parse and return the attested credential data and its length.
     ///
     /// This is assumed to take place after the first 37 bytes of `data`, which is always of fixed size.
     /// - SeeAlso: [WebAuthn Level 3 Editor's Draft §6.5.1. Attested Credential Data]( https://w3c.github.io/webauthn/#sctn-attested-credential-data)
-    private static func parseAttestedData(_ data: [UInt8]) throws -> (AttestedCredentialData, Int) {
+    private static func parseAttestedData(_ data: [UInt8]) throws(WebAuthnError) -> (AttestedCredentialData, Int) {
         /// **aaguid** (16): The AAGUID of the authenticator.
         guard let aaguid = AAGUID(bytes: data[37..<(37 + AAGUID.size)])  // Bytes [37-52]
-        else { throw WebAuthnError.attestedCredentialDataMissing }
+        else { throw .attestedCredentialDataMissing }
 
         /// **credentialIdLength** (2): Byte length L of credentialId, 16-bit unsigned big-endian integer. Value MUST be ≤ 1023.
         let idLengthBytes = data[53..<55]  // Length is 2 bytes
         let idLengthData = Data(idLengthBytes)
         let idLength = UInt16(bigEndianBytes: idLengthData)
 
         guard idLength <= 1023
-        else { throw WebAuthnError.credentialIDTooLong }
+        else { throw .credentialIDTooLong }
 
         let credentialIDEndIndex = Int(idLength) + 55
         guard data.count >= credentialIDEndIndex 
-        else { throw WebAuthnError.credentialIDTooShort }
+        else { throw .credentialIDTooShort }
 
         /// **credentialId** (L): Credential ID
         let credentialID = data[55..<credentialIDEndIndex]
 
         /// **credentialPublicKey** (variable): The credential public key encoded in `COSE_Key` format, as defined in [Section 7](https://tools.ietf.org/html/rfc9052#section-7) of [RFC9052], using the CTAP2 canonical CBOR encoding form.
-        /// Assuming valid CBOR, verify the public key's length by decoding the next CBOR item.
+        /// Assuming valid CBOR, verify the public key's length by decoding the next CBOR item, and checking how much data is left on the stream.
         let inputStream = ByteInputStream(data[credentialIDEndIndex...])
-        let decoder = CBORDecoder(stream: inputStream, options: CBOROptions(maximumDepth: 16))
-        _ = try decoder.decodeItem()
+        do {
+            let decoder = CBORDecoder(stream: inputStream, options: CBOROptions(maximumDepth: 16))
+            _ = try decoder.decodeItem()
+        } catch { throw .invalidPublicKeyLength }
```

Comment:
Took a chance to define a new error case for this and clarify what is happening.